app/src/main/java/com/thermatk/android/xf/fakegapps/FakeSignatures.java

```diff
@@ -12,30 +13,39 @@
 
 public class FakeSignatures implements IXposedHookZygoteInit {
     @Override
-    public void initZygote(StartupParam startupParam) throws Throwable {
-
-        final Class<?> PackageParser = XposedHelpers.findClass("android.content.pm.PackageParser", null);
-        XposedBridge.hookAllMethods(PackageParser, "generatePackageInfo", new XC_MethodHook() {
+    public void initZygote(StartupParam startupParam) {
+        final Class<?> ApplicationPackageManager = XposedHelpers.findClass("android.app.ApplicationPackageManager", null);
+        XposedBridge.hookAllMethods(ApplicationPackageManager, "getPackageInfo", new XC_MethodHook() {
             @Override
-            protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+            protected void afterHookedMethod(MethodHookParam param) {
                 PackageInfo pi = (PackageInfo) param.getResult();
                 if (pi != null) {
                     String packageName = pi.packageName;
-                    if (packageName.contains("com.google.android.gms") || packageName.contains("com.android.vending")) {
-
+                    if (packageName.equals("com.google.android.gms") || packageName.equals("com.android.vending")) {
                         try {
-                            PackageNameServiceManager manager = PackageNameServiceManager.getService();
-                            String callerPackageName = manager.getPackageName(Binder.getCallingUid());
+                            String callerPackageName = "unknown package name";
+                            String callerAppName = "unknown package label";
+
+                            PackageManager pm;
+                            try {
+                                pm = AndroidAppHelper.currentApplication().getApplicationContext().getPackageManager();
+                                callerPackageName = pm.getNameForUid(Binder.getCallingUid());
 
-                            XposedBridge.log("FakeGApps: returning fake signature to " + callerPackageName);
+                                try {
+                                    callerAppName = (String) pm.getApplicationLabel(pm.getApplicationInfo(callerPackageName, 0));
+                                } catch (final PackageManager.NameNotFoundException ignored) {
+                                }
+                            } catch (Exception ignored) {
+                            }
+
+                            XposedBridge.log("FakeGApps: returning fake signature to \"" + callerAppName + "\" (" + callerPackageName + ")");
                         } catch (NullPointerException e) {
-                            XposedBridge.log("FakeGApps: caught the NullPointerException");
+                            XposedBridge.log("FakeGApps: " + e);
                         }
```

Comment:
This `catch` block is useless since a NPE can never happen when concatenating strings, and everything else that might throw it is already wrapped in the nested `try`.